camelot/view/action_steps/item_view.py

```diff
@@ -117,9 +120,9 @@ def __init__( self, admin, value ):
         self.right_toolbar_actions = admin.get_list_toolbar_actions(Qt.RightToolBarArea)
         self.top_toolbar_actions = admin.get_list_toolbar_actions(Qt.TopToolBarArea)
         self.bottom_toolbar_actions = admin.get_list_toolbar_actions(Qt.BottomToolBarArea)
-        self.list_action = admin.get_list_action()
+        self.list_action = AdminRoute._register_list_action_route(self.admin_route, admin.get_list_action())
```

Comment:
Is het gebruik van die AdminRoute hier correct?

Ik heb nu  gelijkaardige code voor die andere acties, maar de camelot/test/test_action.py unit test faalt nu omdat bepaalde routes meerdere malen geregistreerd worden. De individuele cases in test_action.py uitvoeren werkt echter wel. Moeten we eventueel nog een mechanisme toevoegen om geregistreerde routes terug op te ruimen?

Reply:
ik zou geneigd zijn van die register call in de method 'admin.get_list_action()' te laten gebeuren, en die een route te laten teruggeven.  op die manier kunnen we de rest van de code toch al een beetje 'shielden' voor die dubbele registraties.

het opruimen van die geregistreerde routes is een complex vraagstuk.  indien mogelijk zou ik dat tot nu toe trachten te vermijden en trachten de code zo te organiseren dat dit eigenlijk niet nodig is.  ik weet niet of we die lijn overal gaan kunnen doortrekken. maar ik zou het in elk geval in deze fase toch willen proberen.

Reply:
Moet die registratie dan niet gebeuren in de admin constructor?